dotnet/samples/Concepts/ChatCompletion/OpenAI_FunctionCalling.cs

```diff
@@ -94,6 +114,26 @@ public sealed class HolidayRequest
         public string EndDate { get; set; } = string.Empty;
     }
 
+    public class LightPlugin
+    {
+        public bool IsOn { get; set; } = false;
+
+#pragma warning disable CA1024 // Use properties where appropriate
+        [KernelFunction]
+        [Description("Gets the state of the light.")]
+        public string GetState() => IsOn ? "on" : "off";
+#pragma warning restore CA1024 // Use properties where appropriate
```

Comment:
Make this private to avoid the pragmas below?

```suggestion
        private bool _isOn = false;

        [KernelFunction]
        [Description("Gets the state of the light.")]
        public string GetState() => _isOn ? "on" : "off";
```